Update linking-a-pull-request-to-an-issue.md

casals opened the pull request:
### Why:

Fixes #165 .

### What's being changed:

I added a Note box at the top of the document, and a reference to it at the top of the document - should be enough warning without disrupting the rest of the text. Also changed the introductory line in the "Linking a pull request(...)" section as suggested by the author/PR reviewer:

- Original sentence: "You can link a pull request to an issue by using a supported keyword in the pull request's description."
- Modified sentence: "You can link a pull request to an issue by using a supported keyword in the pull request's description or in a commit message (please note that the PR must be on the default branch)."


### Check off the following:
- [x] All of the tests are passing.
- [x] I have reviewed my changes in staging.
- [x] For content changes, I have reviewed the [localization checklist](https://github.com/github/docs/blob/main/contributing/localization-checklist.md)
- [x] For content changes, I have reviewed the [Content style guide for GitHub Docs](https://github.com/github/docs/blob/main/contributing/content-style-guide.md).

casals committed:
Update linking-a-pull-request-to-an-issue.md

Fixes #165 . I added a Note box at the end of the document, and a reference to it at the top of the document - should be enough warning without disrupting the rest of the text. Also changed the introdutory line in the "Linking a pull request(...)" section as suggested by the author.

janiceilene commented:
Amazing @casals! I'll send this one to @github/docs-content-core team to review, too 👍

casals committed:
Update linking-a-pull-request-to-an-issue.md

Included callout tag for note. Moved note to the beginning of the text and removed the related observation in the next paragraph since it became redundant.

janiceilene commented:
Thanks @casals! I'll let the @github/docs-content-core team know this is ready for review 🌟

atlslscsrv-app reviewed (approved):
@atlslscsrv-app

myarb committed:
Merge branch 'main' into patch-10

felicitymay reviewed (commented):
Hi - thanks for addressing this issue, it's great to have this topic improved ✨ 

I'm a little puzzled by:

> I added a Note box at the end of the document, and a reference to it at the top of the document

On [staging](https://docs-199--patch-10.herokuapp.com/en/free-pro-team@latest/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword), the note box appears to be at the start of the topic. Given that our search links to each H3 within the topic, I'm wondering if the note would be better placed within the "[Linking a pull request to an issue using a keyword](https://docs-199--patch-10.herokuapp.com/en/free-pro-team@latest/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)" section, maybe straight after the bulleted list of special terms. Otherwise people who go straight to that section will miss the note. What do you think?

If we did move the note box, then probably we should expand the first sentence of the topic: *"You can link an issue to a pull request manually or using a supported keyword in the pull request description."* to mention the default branch.

casals commented:
Hey @felicitymay - what happened here was: I had just submitted a number of PRs that involved including a Note box at the end of the document. During the review of the first of these PRs, one of the requested changes was to move the Note box to the top of the page, since that's the standard style/practice. In order to maintain consistency, I moved the (new) Note box to the top of the document in all of my PRs - but forgot to change the description here, sorry.

All that being said, I think we can go with your latter suggestion: keep the box at the topic of the document + expand the first sentence of the topic. In fact, this sentence was already extended from its original version. I'm updating the PR description with these changes.

casals committed:
Update linking-a-pull-request-to-an-issue.md

Updated with requested changes

felicitymay commented on content/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue.md:
```diff
@@ -37,7 +43,7 @@ You can manually link up to ten issues to each pull request. The issue and pull
 ### Linking a pull request to an issue using a keyword
 
-You can link a pull request to an issue by using a supported keyword in the pull request's description.
+You can link a pull request to an issue by using a supported keyword in the pull request's description or in a commit message (please note that the PR must be on the default branch). 
```
This looks like a good solution 😄 Thanks 🙇🏻‍♀️ 

We generally write out "pull request" in full (although I can see a few topics do use PR).

```suggestion
You can link a pull request to an issue by using a supported keyword in the pull request's description or in a commit message (please note that the pull request must be on the default branch). 
```

casals committed:
Update content/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue.md

Updated with requested changes

Co-authored-by: Felicity Chapman <felicitymay@github.com>

felicitymay reviewed (approved):
Thanks 👍🏻 🚀

felicitymay committed:
Merge branch 'main' into patch-10

felicitymay commented:
Thanks for responding so quickly - it's much appreciated. I've added the `automerge` label and will check back to make sure that all's well.

felicitymay commented:
Ah. Okay. Will check back to see when the tests have passed and merge.
